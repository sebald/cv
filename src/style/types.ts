import { Interpolation, ThemeProps } from 'styled-components';
import { theme } from './theme';

export type Theme = typeof theme;

export type ThemedInterpolation = Interpolation<ThemeProps<Theme>>;
