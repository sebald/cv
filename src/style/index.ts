export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
