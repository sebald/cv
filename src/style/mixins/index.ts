export * from './animation';
export * from './breakpoints';
export * from './color';
export * from './space';
export * from './typography';
export * from './width';

export * from './types';
