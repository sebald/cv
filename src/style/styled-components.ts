import * as styledComponents from 'styled-components';
import {
  ThemedStyledComponentsModule,
  ThemedStyledFunction,
} from 'styled-components';

import { Theme } from './types';

// https://github.com/styled-components/styled-components/issues/630#issuecomment-317172454
function withProps<U>() {
  return <P, T, O>(
    fn: ThemedStyledFunction<P, T, O>
  ): ThemedStyledFunction<P & U, T, O & U> =>
    fn as ThemedStyledFunction<P & U, T, O & U>;
}

// Alternative, maybe better approach?
type TTag = keyof JSX.IntrinsicElements;

function suit<U>(tag: TTag) {
  return styled[tag] as ThemedStyledFunction<
    JSX.IntrinsicElements[TTag] & U,
    Theme
  >;
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export { styled, css, injectGlobal, keyframes, withProps, suit, ThemeProvider };
