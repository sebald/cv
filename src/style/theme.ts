export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
