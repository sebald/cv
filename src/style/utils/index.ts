export * from './classnames';
export * from './constants';
export * from './createMixinComponent';
export * from './interpolation';
export * from './unit';
export * from './typography';
