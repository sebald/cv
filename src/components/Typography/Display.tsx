import { styled, typography } from '../../style';

export const Display = styled.h1`
  ${typography('display')};
`;
Display.displayName = 'Display';
