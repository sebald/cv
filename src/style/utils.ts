import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
