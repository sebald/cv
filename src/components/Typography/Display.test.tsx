import { shallow } from 'enzyme';
import * as React from 'react';

import { theme } from '../../style';
import { Display } from './Display';

test('render correctly', () => {
  const wrapper = shallow(<Display theme={theme}>Text</Display>);
  expect(wrapper).toMatchSnapshot();
});
