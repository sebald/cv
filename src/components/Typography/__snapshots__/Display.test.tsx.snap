// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
.c0 {
  font-size: 3.375rem;
  line-height: 4rem;
  padding-top: 0.75rem;
  margin-top: 0;
  margin-bottom: 1.25rem;
  color: #212121;
  font-family: 'Montserrat',sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

<h1
  className="c0"
>
  Text
</h1>
`;
